chronos/asyncfutures2.nim

```diff
@@ -480,6 +538,11 @@ proc all*[T](futs: varargs[Future[T]]): auto =
   ##
   ## Note, that if one of the futures in ``futs`` will fail, result of ``all()``
   ## will also be failed with error from failed future.
+  ##
+  ## TODO: This procedure has bug on handling cancelled futures from ``futs``.
```

Comment:
This `all()` is going to be deprecated because it is logically broken. From my point of view all the aggregation procedures which are returning actual values are logically broken. Because it hard to understand what you need to return as aggregated data value, when one of the futures got failed or cancelled.

Reply:
`all` is synonymous to `and` to me. So ideally it should return a corresponding tuple (i.e. `proc all(f1: Future[T1], f2: Future[T2], ...): (T1, T2, ...)`) if `all` futures are completed, and fail if at least one of the futures has failed. That said, I see it more as a metaprogramming exercise rather than something really useful :)